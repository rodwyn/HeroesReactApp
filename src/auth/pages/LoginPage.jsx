import { useNavigate } from "react-router-dom"

export const LoginPage = () => {
  const navigate = useNavigate();

  const onLoging = ()=> {
    navigate('/', {
      replace: true
    })
  };

  return (
    <div className="container mt-5">
      <h1>Login</h1>
      <hr />

      <button
        className="btn btn-primary"
        onClick={onLoging}
      >
        Login
      </button>
    </div>
  )
}
