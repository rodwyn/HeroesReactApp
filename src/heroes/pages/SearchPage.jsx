import { useLocation, useNavigate } from "react-router-dom";
import queryString from "query-string";
import { useForm } from "../../hooks/useForm"
import { HeroCard } from "../components/HeroCard"
import { getHeroesByName } from "../helpers/getHeroesByName";

export const SearchPage = () => {
  const navigate = useNavigate();
  const location = useLocation();
  const { q = '' } = queryString.parse(location.search);
  const heroes = getHeroesByName(q);

  const {searchText, onInputChange} = useForm({
    searchText: q
  });

  const onSearchSubmit = event => {
    event.preventDefault();
    // if (searchText.trim().length <= 0) return;

    navigate(`?q=${searchText}`);
  };

  return (
    <>
      <h1>SearchPage</h1>
      <hr />

      <div className="row">
        <div className="col-5">
          <h4>Searching</h4>
          <hr />
          <form onSubmit={onSearchSubmit}>
            <input
              aria-label="search"
              type="text"
              placeholder="search a hero"
              className="form-control"
              name="searchText"
              autoComplete="off"
              value={searchText}
              onChange={onInputChange}
            />
            <button className="btn btn-outline-primary mt-1">
              Search
            </button>
          </form>
        </div>
        <div className="col-7">
          <h4>Results</h4>
          <hr />

          {
            (q === '') 
            ? <div className="alert alert-primary animate__animated animate__fadeIn">Search a hero</div>
            : (heroes.length === 0)  && <div aria-label="not-hero" className="alert alert-danger animate__animated animate__fadeIn">Not hero with <b>{q}</b></div>
          }

          {
            heroes.map(hero => <HeroCard key={hero.id} {...hero} />)
          }

        </div>
      </div>
    </>
  )
}
