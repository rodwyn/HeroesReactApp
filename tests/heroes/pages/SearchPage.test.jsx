import { fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { SearchPage } from "../../../src/heroes/pages/SearchPage";

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: ()=> mockedUseNavigate
}));

beforeEach(() => jest.clearAllMocks());

describe('Testing <SearchPage />.', () => {
  test('should render correctly with default values.', () => {
    const {container} = render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    )

    expect(container).toMatchSnapshot();
  });

  test('should render batman card.', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=batman']}>
        <SearchPage />
      </MemoryRouter>
    )

    const input = screen.getByRole('textbox');
    const img = screen.getByRole('img');
    expect(input.value).toBe('batman');
    expect(img.src).toContain('/assets/heroes/dc-batman.jpg');
  });

  test('should render alert on none result.', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=xyx']}>
        <SearchPage />
      </MemoryRouter>
    )

    const alert = screen.getByLabelText('not-hero');
    expect(alert).toBeTruthy();
  });

  test('should call navigate to hero page.', () => {
    render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    )
    
    const input = screen.getByLabelText('search');
    fireEvent.change(input, {target: { name: 'searchText', value: 'batman'}})
    fireEvent.submit(input)
    expect(mockedUseNavigate).toHaveBeenCalledWith('?q=batman');
  });
});