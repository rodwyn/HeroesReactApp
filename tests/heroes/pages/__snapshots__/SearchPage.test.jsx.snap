// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <SearchPage />. should render correctly with default values. 1`] = `
<div>
  <h1>
    SearchPage
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form>
        <input
          aria-label="search"
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search a hero
      </div>
    </div>
  </div>
</div>
`;
